src/index.spec.ts

```diff
@@ -0,0 +1,323 @@
+    }, {
+      year: 2021,
+      depreciationMonths: 11,
+      depreciationAmount: 366667,
+      percentage: 0.1833335,
+      startAmount: 366667,
+      endAmount: 0,
+    }])
+  });
+
+  it("should handle any month, besides January or December", () => {
+    expect(calculateDepreciation({
+      purchaseAmount: 2000000,
+      purchaseDate: new Date("2016-08-01"),
+      totalDepreciationYears: 5,
+    })).toEqual([{
+      year: 2016,
+      depreciationMonths: 5,
+      depreciationAmount: 166667,
+      percentage: 0.0833335,
+      startAmount: 2000000,
+      endAmount: 1833333,
+    }, {
+      year: 2017,
+      depreciationMonths: 12,
+      depreciationAmount: 400000,
+      percentage: 0.2,
+      startAmount: 1833333,
+      endAmount: 1433333,
+    }, {
+      year: 2018,
+      depreciationMonths: 12,
+      depreciationAmount: 400000,
+      percentage: 0.2,
+      startAmount: 1433333,
+      endAmount: 1033333,
+    }, {
+      year: 2019,
+      depreciationMonths: 12,
+      depreciationAmount: 400000,
+      percentage: 0.2,
+      startAmount: 1033333,
+      endAmount: 633333,
+    }, {
+      year: 2020,
+      depreciationMonths: 12,
+      depreciationAmount: 400000,
+      percentage: 0.2,
+      startAmount: 633333,
+      endAmount: 233333,
+    }, {
+      year: 2021,
+      depreciationMonths: 7,
+      depreciationAmount: 233333,
+      percentage: 0.1166665,
+      startAmount: 233333,
+      endAmount: 0,
+    }])
+  });
+
+  it("should handle one year depreciation - purchase made in Jan", () => {
+    expect(calculateDepreciation({
+      purchaseAmount: 2000000,
+      purchaseDate: new Date("2016-01-01"),
+      totalDepreciationYears: 1,
+    })).toEqual([{
+      year: 2016,
+      depreciationMonths: 12,
+      depreciationAmount: 2000000,
+      percentage: 1,
+      startAmount: 2000000,
+      endAmount: 0,
+    }])
+  });
+
+  it("should handle one year depreciation - purchase made after Jan", () => {
+    expect(calculateDepreciation({
+      purchaseAmount: 2000000,
+      purchaseDate: new Date("2016-05-01"),
+      totalDepreciationYears: 1,
+    })).toEqual([{
+      year: 2016,
+      depreciationMonths: 8,
+      depreciationAmount: 1333333,
+      percentage: 0.6666665,
+      startAmount: 2000000,
+      endAmount: 666667,
+    }, {
+      year: 2017,
+      depreciationMonths: 4,
+      depreciationAmount: 666667,
+      percentage: 0.3333335,
+      startAmount: 666667,
+      endAmount: 0,
+    }])
+  });
+
+  it("should handle the case where good's value is less than 80000 cents", () => {
+    expect(calculateDepreciation({
+      purchaseAmount: 79900,
+      purchaseDate: new Date("2016-05-01"),
+      totalDepreciationYears: 5,
+    })).toEqual([{
+      year: 2016,
+      depreciationMonths: 8,
+      depreciationAmount: 79900,
+      percentage: 1,
+      startAmount: 79900,
+      endAmount: 0,
+    }])
+  });
+
+  it("should not return endAmount of 1", () => {
```

Comment:
❓ Can you please clarify this test step?

Reply:
I added a comment here: https://github.com/kontist/asset-depreciation-calculator/pull/1/files#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80R52